core/parsigex/parsigex.go

```diff
@@ -104,6 +104,14 @@ func (m *ParSigEx) Broadcast(ctx context.Context, duty core.Duty, set core.ParSi
 		return err
 	}
 
+	// TODO: remove this from prod once we know what's going on
+	for _, data := range pb.Set {
+		_, err = core.ParSignedDataFromProto(duty.Type, data)
```

Comment:
what about doing this inside `ParSignedDataSetToProto`?

Also, rather please use `ParSignedDataSetFromProto`

Reply:
> what about doing this inside ParSignedDataSetToProto?

I need to pass duty type to this function, I would rather use `core.ParSignedDataSetFromProto` as suggested.